api.js

```diff
@@ -99,6 +100,25 @@ Api.prototype._handleTest = function (test) {
 	this.emit('test', test);
 };
 
+Api.prototype._findBase = function (files) {
+	this.base = files.reduce(function (base, file) {
+		file = path.relative('.', file);
+		file = file.split(path.sep);
+		if (base === false) {
+			return file;
+		}
+		return base.filter(function (part, i) {
+			return file[i].toLowerCase() === part.toLowerCase();
+		});
+	}, false).join(path.sep);
+
+	if (this.base === '' || this.base === '.') {
+		this.base = this.files[0] || 'test';
+	}
+
+	this.base += path.sep;
+};
```

Comment:
Naturally, I spend 30 minutes on something to find out that there's already something that does that.